help_to_heat/frontdoor/views.py

```diff
@@ -635,11 +648,14 @@ def save_post_data(self, data, session_id, page_name):
                     data[address_choice_field] = address_choice_field_write_address
                 else:
                     data[address_choice_field] = address_choice_field_epc_api_fail
+                    check_address_has_results_in_address_api()
             else:
                 data[address_choice_field] = address_choice_field_write_address
+                check_address_has_results_in_address_api()
         except Exception as e:  # noqa: B902
             logger.exception(f"An error occurred: {e}")
             data[address_choice_field] = address_choice_field_epc_api_fail
+            check_address_has_results_in_address_api()
```

Comment:
Doing this feels brittle. I think I'd rather have one call after the try-catch. I don't follow why we don't call it in the one branch - why's that?

Reply:
in some cases we don't care for what the OS API response is, the only case being if the epc api returns results. could rewrite the logic to include just one call by checking afterwards